wandb/sdk/wandb_init.py

```diff
@@ -859,23 +859,36 @@ def init(  # noqa: C901
             f"\nconfig: {config.base_no_artifacts}"
         )
 
-        if wandb.run is not None and os.getpid() == wandb.run._init_pid:
```

Comment:
could you explain the change in the condition? why is it safe to remove the pid?

Reply:
The PID check happens when accessing the singleton (`_wl`). If we're in a new process, we'll have a fresh singleton.

So if `self._wl.most_recent_active_run` is set, that implies it was created (or attached to) by the current process.

Reply:
should we leave a comment here?

Reply:
I'd like it to be unnecessary to think about at all; this PID checking logic was duplicated in a couple places, but I think we can make it the responsibility of the singleton mechanism. Basically anything the singleton returns should be assumed to be set up appropriately for the current process.

Reply:
fair point! sounds good